src/app/workspace/resource/representation/video/video.component.ts

```diff
@@ -0,0 +1,364 @@
+     * video navigation from button (-/+ 10 sec)
+     *
+     * @param range positive or negative number value
+     */
+    updateTimeFromButton(range: number) {
+        if (range > 0 && this.currentTime > (this.duration - 10)) {
+            this._navigate(this.duration);
+        } else if (range < 0 && this.currentTime < 10) {
+            this._navigate(0);
+        } else {
+            this._navigate(this.currentTime + range);
+        }
+    }
+    /**
+     * video naviagtion from timeline / progress bar
+     *
+     * @param ev MatSliderChange
+     */
+    updateTimeFromSlider(time: number) {
+        this._navigate(time);
+    }
+    /**
+     * video navigation from scroll event
+     *
+     * @param ev WheelEvent
+     */
+    updateTimeFromScroll(ev: WheelEvent) {
+        ev.preventDefault();
+        this._navigate(this.currentTime + (ev.deltaY / 25));
+    }
+
+
+    /**
+     * event on mouses move on timeline
+     * @param ev MouseEvent
+     */
+    mouseMove(ev: MouseEvent) {
+        this._calcPreviewTime(ev);
+    }
+
+    /**
+     * show preview image during "mousemove" on progress bar / timeline
+     *
+     * @param  ev PointerValue
+     */
+    updatePreview(ev: PointerValue) {
+
+        this.displayPreview(true);
+
+        this.previewTime = Math.round(ev.time);
+
+        // position from left:
+        let leftPosition: number = (ev.position - this.timelineDimension.x) - this.halfFrameWidth;
+
+        // prevent overflow of preview image on the left
+        if (leftPosition <= 8) {
+            leftPosition = 8;
+        }
+
+        // prevent overflow of preview image on the right
+        if (leftPosition >= (this.timelineDimension.width - this.frameWidth + 8)) {
+            leftPosition = this.timelineDimension.width - this.frameWidth + 8;
+        }
+
+        // set preview positon on x axis
+        this.preview.nativeElement.style.left = leftPosition + 'px';
+
+    }
+
+    /**
+     * show preview image or hide it
+     *
+     * @param status true = show ('block'), false = hide ('none')
+     */
+    displayPreview(status: boolean) {
+        this.preview.nativeElement.style.display = (status ? 'block' : 'none');
+    }
+
+    /**
+     * opens replace file dialog
+     *
+     */
+    openReplaceFileDialog() {
+        const propId = this.parentResource.properties[Constants.HasMovingImageFileValue][0].id;
+
+        const dialogConfig: MatDialogConfig = {
+            width: '800px',
+            maxHeight: '80vh',
+            position: {
+                top: '112px'
+            },
+            data: { mode: 'replaceFile', title: 'Video (mp4)', subtitle: 'Update the video file of this resource', representation: 'movingImage', id: propId },
+            disableClose: true
+        };
+        const dialogRef = this._dialog.open(
+            DialogComponent,
+            dialogConfig
+        );
+
+        dialogRef.afterClosed().subscribe((data) => {
+            this._replaceFile(data);
```

Comment:
We should check if `data` is undefined here, otherwise it breaks the video player. This is my fault actually and should be changed in each `openReplaceFileDialog` that I added in #695. Interestingly this didn't cause an issue with the other `openReplaceFileDialog` methods which is why I didn't catch it. Should I open a separate PR for this once this is merged?

Reply:
We should do it in a separate task. 
I can disable the "replace-file" button in this task to avoid crashing player.

Reply:
resolved in a6527d44